src/neuroconv/tools/nwb_helpers/_models/_zarr_models.py

```diff
@@ -120,3 +122,26 @@ def get_data_io_keyword_arguments(self) -> Dict[str, Any]:
             compressor = False
 
         return dict(chunks=self.chunk_shape, filters=filters, compressor=compressor)
+
+
+class ZarrBackendConfiguration(BackendConfiguration):
+    """A model for matching collections of DatasetConfigurations specific to the Zarr backend."""
+
+    backend: Literal["zarr"] = Field(
+        default="zarr", description="The name of the backend used to configure the NWBFile."
+    )
+    data_io_class: Type[ZarrDataIO] = Field(
+        default=ZarrDataIO, description="The DataIO class that is specific to Zarr."
+    )
+    dataset_configurations: Dict[str, ZarrDatasetConfiguration] = Field(
+        description=(
+            "A mapping from object locations to their ZarrDatasetConfiguration specification that contains all "
+            "information for writing the datasets to disk using the Zarr backend."
+        )
+    )
+    number_of_jobs: int = Field(
+        description="Number of jobs to use in parallel during write.",
+        ge=-psutil.cpu_count(),  # TODO: should we specify logical=False in cpu_count?
+        le=psutil.cpu_count(),
+        default=-2,  # -2 translates to 'all CPU except for one'
```

Comment:
In the case someone has 4 logical cores this could be:

`-4, -3, -2, -1, 0, 1, 2, 3, 4` 

Questions:
What do negative numbers mean?
Why is -2 equivalent to 3? as per your comment in the default?
What does 0 mean?
 
is this a convention from somewhere?

Reply:
TODO: need to document negative slicing better